akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -1125,7 +1125,7 @@ private[stream] final class Delay[T](d: FiniteDuration, strategy: DelayOverflowS
           case Backpressure ⇒ throw new IllegalStateException("Delay buffer must never overflow in Backpressure mode")
         }
         else {
-          grabAndPull(strategy != Backpressure || buffer.capacity < size - 1)
```

Comment:
But this was not the main issue, the issue was the `inputBuffer` attribute among the initial attributes which basically disabled overriding it properly.